compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/methodgen/MainMethodGen.java

```diff
@@ -232,9 +239,75 @@ private void generateExecuteFunctionCall(String initClass, MethodVisitor mv, BIR
     }
 
     private void generateModuleStopCall(String initClass, MethodVisitor mv) {
+        String lambdaName = LAMBDA_PREFIX + "stopdynamic";
+
+        mv.visitTypeInsn(NEW, SCHEDULER);
+        mv.visitInsn(DUP);
+        mv.visitInsn(ICONST_1);
+        mv.visitInsn(ICONST_0);
+        mv.visitMethodInsn(INVOKESPECIAL, SCHEDULER, JVM_INIT_METHOD, "(IZ)V", false);
+        mv.visitVarInsn(ASTORE, indexMap.addIfNotExists("newSchedulerVar", symbolTable.anyType));
+
+        generateCallStopDynamicLambda(mv, lambdaName, initClass, asyncDataCollector);
+
+        mv.visitVarInsn(ALOAD, indexMap.get("newSchedulerVar"));
+        mv.visitVarInsn(ALOAD, indexMap.get(FUTURE_VAR));
+        mv.visitMethodInsn(INVOKESTATIC, initClass, MODULE_STOP_METHOD, MODULE_STOP_METHOD_DESC, false);
+    }
+
+    private void generateCallStopDynamicLambda(MethodVisitor mv, String lambdaName, String moduleInitClass,
+                                               AsyncDataCollector asyncDataCollector) {
+        addRuntimeRegistryAsParameter(mv);
+        int futureIndex = indexMap.addIfNotExists(FUTURE_VAR, symbolTable.anyType);
+        generateMethodBody(mv, moduleInitClass, lambdaName, asyncDataCollector);
+
+        // handle any runtime errors
+        Label labelIf = new Label();
+        mv.visitVarInsn(ALOAD, futureIndex);
+        mv.visitFieldInsn(GETFIELD, FUTURE_VALUE, PANIC_FIELD, GET_THROWABLE);
+        mv.visitJumpInsn(IFNULL, labelIf);
+
+        mv.visitVarInsn(ALOAD, futureIndex);
+        mv.visitFieldInsn(GETFIELD, FUTURE_VALUE, PANIC_FIELD, GET_THROWABLE);
+        mv.visitMethodInsn(INVOKESTATIC, RUNTIME_UTILS, HANDLE_STOP_PANIC_METHOD, HANDLE_STOP_PANIC,
+                false);
+        mv.visitLabel(labelIf);
+    }
+
+    private void addRuntimeRegistryAsParameter(MethodVisitor mv) {
+        int arrIndex = indexMap.addIfNotExists(ARR_VAR, symbolTable.anyType);
+        mv.visitIntInsn(BIPUSH, 2);
+        mv.visitTypeInsn(ANEWARRAY, OBJECT);
+        mv.visitVarInsn(ASTORE, arrIndex);
+        mv.visitVarInsn(ALOAD, arrIndex);
+        mv.visitInsn(ICONST_1);
         mv.visitVarInsn(ALOAD, indexMap.get(SCHEDULER_VAR));
         mv.visitMethodInsn(INVOKEVIRTUAL, SCHEDULER, "getRuntimeRegistry", GET_RUNTIME_REGISTRY_CLASS, false);
-        mv.visitMethodInsn(INVOKESTATIC, initClass, MODULE_STOP_METHOD, INIT_RUNTIME_REGISTRY, false);
+        mv.visitInsn(AASTORE);
+        mv.visitVarInsn(ALOAD, indexMap.get("newSchedulerVar"));
+        mv.visitVarInsn(ALOAD, arrIndex);
+    }
+
+    private void generateMethodBody(MethodVisitor mv, String initClass, String stopFuncName,
+                                    AsyncDataCollector asyncDataCollector) {
+        JvmCodeGenUtil.createFunctionPointer(mv, initClass + "$SignalListener", stopFuncName);
+        mv.visitInsn(ACONST_NULL);
+        jvmTypeGen.loadType(mv, new BNilType());
```

Comment:
Can we use constant new BNilType()

Reply:
Used the `mv.visitFieldInsn(GETSTATIC, PREDEFINED_TYPES, "TYPE_NULL", LOAD_NULL_TYPE)` instead `jvmTypeGen.loadType(mv, new BNilType())` in 8da28c04c1397a39613ef773b0f56c5989ce3016. Could you please verify it?